roles/openshift_facts/library/openshift_facts.py

```diff
@@ -914,12 +914,17 @@ def set_sdn_facts_if_unset(facts, system_facts):
 
             if 'networkConfig' in config:
                 if 'clusterNetworkCIDR' in config['networkConfig']:
-                    facts['master']['sdn_cluster_network_cidr'] = \
+                    sdn_cluster_network_cidr = \
                         config['networkConfig']['clusterNetworkCIDR']
                 if 'hostSubnetLength' in config['networkConfig']:
-                    facts['master']['sdn_host_subnet_length'] = \
+                    sdn_host_subnet_length = \
                         config['networkConfig']['hostSubnetLength']
 
+        if 'sdn_cluster_network_cidr' not in facts['master']:
```

Comment:
This will allow vars to be overridden but the config file values will never overwrite user provided because they will already be set.

I think that the config section should set directly to facts (overwriting) and the defaults should be set when no value is already set.

```diff
         # set defaults for sdn_cluster_network_cidr and sdn_host_subnet_length
         # these might be overridden if they exist in the master config file
-        facts['master']['sdn_cluster_network_cidr'] = '10.128.0.0/14'
-        facts['master']['sdn_host_subnet_length'] = '9'
+        if 'sdn_cluster_network_cidr' not in facts['master']:
+            facts['master']['sdn_cluster_network_cidr'] = '10.128.0.0/14'
+        if 'sdn_host_subnet_length' not in facts['master']:
+            facts['master']['sdn_host_subnet_length'] = '9'
```